Working with blob storage nb

apreshill opened the pull request.

apreshill committed:
draft

apreshill committed:
first draft

pierrebrunelle reviewed (commented):
The notebook uses `t.source_image.flip('horizontal')` but there is no `flip` method on Image type in Pixeltable.

Looking at `pixeltable/functions/image.py`, the available method for flipping is `transpose(method)` where:
- `0` = FLIP_LEFT_RIGHT (horizontal)
- `1` = FLIP_TOP_BOTTOM (vertical)

The fix:
```python
t.add_computed_column(
    flipped=t.source_image.transpose(0),  # horizontal flip
    destination=str(local_dest / 'flipped'),
)
```

Also update the example code blocks in the markdown cells that reference `flip()`.

apreshill committed:
revamped the nb

apreshill committed:
Merge remote-tracking branch 'upstream/main' into blob-nb

apreshill committed:
changes for flow

apreshill committed:
more zsushing

pierrebrunelle committed:
Merge branch 'main' into blob-nb

sergey-mkhitaryan committed:
Fix a bug in rag-demo.ipynb (#996)

1. In the notebook, fix the column that is being recomputed in the end.
2. Restore notebook outputs
3. Fix a pdfium warning by calling a different function.

apreshill committed:
Fixes the errant `/datastore/` url in the Reve docstrings (#999)

You can view the existing link here:
https://docs.pixeltable.com/sdk/latest/reve

aaron-siegel committed:
Remove custom-iterators.ipynb from docs for now, and clean up docs.json (#997)

sergey-mkhitaryan committed:
[PXT-921] Skip test_create_video_table on cockroachdb (#1002)

pierrebrunelle committed:
Add iterators cookbook with all 6 built-in iterators (#1000)

## Summary

Add a cookbook demonstrating all 6 built-in iterator functions for
splitting data into multiple rows.

- **Updated `iterators.mdx`:**
  - Replace deprecated `paragraph` separator with `sentence,token_limit`
  - Add links to related cookbooks

- **Updated `docs.json`:**
  - Add new cookbook to Core Operations section

aaron-siegel committed:
Update nightly.yml

amithadke committed:
PXT 910 Add rerun options to presigned url tests (#1006)

pierrebrunelle committed:
docs: add presigned_url to S3 cookbook and update SDK docs (#1004)

Updates documentation to reflect the new `presigned_url` feature from PR
#991.

 **SDK Documentation Update** (`public_api.opml`)
   - Added `pixeltable.functions.net` module with `presigned_url` UDF
- Added `pixeltable.functions.uuid` module with `uuid4` UDF (was in
codebase but missing from docs)

Xe committed:
docs(providers): add Tigris example notebook (#998)

A small tutorial to get people started with Tigris and Pixeltable.

---------

Signed-off-by: Xe Iaso <xe@tigrisdata.com>
Co-authored-by: Pierre Brunelle <pierre@noteable.io>
Co-authored-by: Pierre Brunelle <70675979+pierrebrunelle@users.noreply.github.com>

pierrebrunelle committed:
docs: update Mintlify theme colors and styling (#1008)

## Summary
- Add custom `style.css` with website-matching theme
- Update card icons to use duotone style for visual consistency
- Small fixes on intro page
- Fix callout colors to use blue instead of default green
- Update fonts (DM Sans for headings, Inter for body)
- Added menu to copy to markdown and talk to LLMs
- Enabled breadcrumb and fixed SEO

<img width="1436" height="1047" alt="image"
src="https://github.com/user-attachments/assets/3f0ab4f0-222f-45f3-b7f6-d0cd09b3d438"
/>

<img width="1444" height="745" alt="image"
src="https://github.com/user-attachments/assets/a02faca8-bb2e-478e-8825-24080ebe35ce"
/>

apreshill committed:
add presigned urls

apreshill committed:
Merge remote-tracking branch 'upstream/main' into blob-nb

apreshill committed:
Merge branch 'main' into blob-nb

apreshill committed:
Use high performance endpoint for Tigris (#1011)

Current docs recommend the Tigris fly endpoint, which is only recommend
for apps running within fly. The general purpose (and higher
performance) endpoint is [t3.storage.dev](http://t3.storage.dev/).

https://docs.pixeltable.com/integrations/cloud-storage#tigris

sergey-mkhitaryan committed:
Merge Table.add_embedding_index examples (#1014)

aaron-siegel committed:
Notebook fixes & some cleanup (#1010)

- Add missing dependencies to several notebooks
- Add ffmpeg install to run-isolated-nb-tests
- Add missing require_package directives to a few UDFs
- Remove obsolete and undocumented `sentence_transformer_list` and
`cross_encoder_list` UDFs (we have `pxtf.map()` now for this)

mkornacker committed:
Progress tracker (#956)

- replaces tqdm progress bars with more detailed progress indicators
using rich
- removes tracking of 'computed expressions', since this doesn't really
map onto anything the user cares about

aaron-siegel committed:
[PXT-925] Fix spurious exception when `if_not_exists='ignore'` is used with a missing parent dir (#1015)

Fixes both `drop_table()` and `drop_dir()`.

aaron-siegel committed:
Improve primary key error message (#1016)

This has bitten a couple users now.

pierrebrunelle committed:
Add files via upload

mkornacker committed:
Bedrock invoke_model() udf (#1018)

pierrebrunelle committed:
[PXT-765] Support for Office Formats as part of Document Type through MarkdownIT (#960)

## Add PowerPoint, Word, and Excel support via MarkItDown

**Core Integration:**
- Added `markitdown[pptx,docx,xlsx]>=0.1.3` dependency
- Extended `DocumentType.DocumentFormat` enum with `PPTX`, `DOCX`,
`XLSX`
- Created `get_office_handle()` that converts office files to markdown
AST via MarkItDown
- Added `_markitdown_sections()` to `DocumentSplitter` for processing
converted documents
- Registered `markitdown` as optional package in `env.py`

**Design Decision:**
Office documents are converted to markdown on load and processed through
the existing markdown AST pipeline. This provides consistent behavior
across all document types and leverages existing splitting/metadata
logic.

**Bug Fix:**
Fixed edge case in `_token_chunks()` where Unicode decode failures could
cause assertion errors on certain document content.

**Testing:**
- Added 3 tests in `test_document.py` covering insert, handle creation,
and splitting
- Tested with 26,971-line CalPy presentation (splits into 2,369 token
chunks)

**Documentation:**
- Created cookbook recipe: `doc-extract-text-from-office-files.ipynb`
- Updated cookbook README, CONTRIBUTING.md, and STYLE_GUIDE.md

### Supported Formats
- PowerPoint: `.pptx`, `.ppt`
- Word: `.docx`, `.doc`  
- Excel: `.xlsx`, `.xls`

aaron-siegel committed:
Update nightly.yml

pierrebrunelle committed:
Add files via upload

pierrebrunelle committed:
Add files via upload

pierrebrunelle committed:
Add files via upload

mkornacker committed:
HF DetrForSegmentation (#1020)

Also included:
- udf overlay_segmentation() in pxtf.vision to visualize the
segmentation map

pierrebrunelle committed:
Image2Image: Updated HF.py to use AutoPipelineForImage2Image and Cookbook (#1025)

- Add cookbook: Transform images with AI-powered editing - Created the
notebook
- feat: Update image_to_image UDF to use AutoPipelineForImage2Image -
Changed from StableDiffusionImg2ImgPipeline to
AutoPipelineForImage2Image enabling other models support **matching our
text-to-image UDF**
- `torch_dtype` is deprecated! Use `dtype` instead`
- docs: Add img-image-to-image cookbook to navigation - Added to
docs.json
- style: Add missing comment to match text_to_image pattern
- Added # Disable safety checker for performance comment for consistency

apreshill committed:
bucket swap

apreshill committed:
Update notebook to use pxt-test bucket

apreshill committed:
Merge remote-tracking branch 'upstream/main' into blob-nb

# Conflicts:
#	docs/release/howto/cookbooks/audio/audio-extract-from-video.ipynb
#	pixeltable/catalog/table.py
#	pixeltable/exec/cache_prefetch_node.py

apreshill committed:
merge in main

apreshill committed:
get presigned urls with buckets working

apreshill committed:
Merge upstream main into blob-nb

apreshill committed:
run make format

apreshill committed:
add html comment about access key required to skip tests
